engines/docker/engine.go

```diff
@@ -0,0 +1,97 @@
+package dockerengine
+
+import (
+	"sync"
+
+	docker "github.com/fsouza/go-dockerclient"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/caching"
+)
+
+type engine struct {
+	engines.EngineBase
+	m           sync.Mutex
+	Environment *runtime.Environment
+	client      *docker.Client
```

Comment:
nit, but could we rename this to `docker  *docker.Client` ? it would look much better... `engine.client` could be anything.